### PR TITLE
Fix/gas refund on set vote

### PR DESCRIPTION
**Pull request by dcrescimbeni**
Relates to issue #159, and DXD-11 in the audit.

Fixes `setVote` exploit when `voteGas` amount is set above the actual gas spent for the function call, enabling voters to get refunded more than what they spent to call the function.

It sets a hard limit of 100.000 gas (~85% of the actual 117.000 gas needed to execute `setVote()`).

### CODE REVIEW

**Comment by rossneilson** [comment]
So we are going with the assumption that gas will be around that cost regardless of chain?

**Comment by dcrescimbeni** [comment]
> So we are going with the assumption that gas will be around that cost regardless of chain?

That's the assumption, yes.

If that's not correct, then gas consumption must be calculated every time the function is called.

**Review by rossneilson — DISMISSED** [review]

**Review by AugustoL — CHANGES_REQUESTED** [review]
Set the limit to 117 000, not 100 000

**Comment by dcrescimbeni** [comment]
@AugustoL done!

**Review by AugustoL — CHANGES_REQUESTED** [review]

File: contracts/erc20guild/BaseERC20Guild.sol [hunk 1]
```diff
@@ -179,7 +179,7 @@ contract BaseERC20Guild {
         require(_proposalTime > 0, "ERC20Guild: proposal time has to be more than 0");
         require(_lockTime >= _proposalTime, "ERC20Guild: lockTime has to be higher or equal to proposalTime");
         require(_votingPowerForProposalExecution > 0, "ERC20Guild: voting power for execution has to be more than 0");
-        require(_voteGas <= 100000, "ERC20Guild: vote gas has to be equal or lower than 100.000"); // 100.000 is 85% of the gas amount to execute setVote
+        require(_voteGas <= 117000, "ERC20Guild: vote gas has to be equal or lower than 117.000");
```
[the remark this answers is not in the document]
Comment (dcrescimbeni) [reply contracts/erc20guild/BaseERC20Guild.sol]: Done!

**Review by AugustoL — APPROVED** [review]

**Review by rossneilson — APPROVED** [review]

**Pull request merged**